chromadb/segment/impl/manager/distributed.py

```diff
@@ -82,33 +84,25 @@ def delete_segments(self, collection_id: UUID) -> Sequence[UUID]:
         "DistributedSegmentManager.get_segment",
         OpenTelemetryGranularity.OPERATION_AND_SEGMENT,
     )
-    def get_segment(self, collection_id: UUID, type: Type[S]) -> S:
-        if type == MetadataReader:
-            scope = SegmentScope.METADATA
-        elif type == VectorReader:
-            scope = SegmentScope.VECTOR
-        else:
-            raise ValueError(f"Invalid segment type: {type}")
-
+    def get_segment(self, collection_id: UUID, scope: SegmentScope) -> Segment:
         if scope not in self._segment_cache[collection_id]:
-            segments = self._sysdb.get_segments(collection=collection_id, scope=scope)
-            known_types = set([k.value for k in SEGMENT_TYPE_IMPLS.keys()])
-            # Get the first segment of a known type
-            segment = next(filter(lambda s: s["type"] in known_types, segments))
-            grpc_url = self._segment_directory.get_segment_endpoint(segment)
-            if segment["metadata"] is not None:
-                segment["metadata"]["grpc_url"] = grpc_url  # type: ignore
-            else:
-                segment["metadata"] = {"grpc_url": grpc_url}
+            # There should be exactly one segment per scope for a collection
```

Comment:
this is for now only, can we say for now

Reply:
Updated comment